pkg/controller/configmap/config.go

```diff
@@ -623,6 +621,24 @@ func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap)
 		if !needChange.agent && inSlice(sec, operatortypes.AgentSections) {
 			needChange.agent = true
 		}
+		if !needChange.agent && inSlice(sec, []string{"k8s", "coe"}) {
```

Comment:
operatortypes.AgentSections are []string{"DEFAULT", "k8s", "coe", "nsx_node_agent", "nsx_kube_proxy"}, which means any changes in one of section AgentSections, needChange.agent will be set true at #L622.  If this, the logic from #L624 won't be processed. 
Here, we only want to check what option changes will make node agent restart. So, we want to restart node agent ds only when changes are in AgentKeys, is it?

Reply:
we may not need code line#L621-622

Reply:
> operatortypes.AgentSections are []string{"DEFAULT", "k8s", "coe", "nsx_node_agent", "nsx_kube_proxy"}, which means any changes in one of section AgentSections, needChange.agent will be set true at #L622. If this, the logic from #L624 won't be processed. Here, we only want to check what option changes will make node agent restart. So, we want to restart node agent ds only when changes are in AgentKeys, is it?

Actually we need line#L621-622, but I forget to change defaults.go. Please check the latest change in defaults.go, in this file, section k8s and coe are removed. This means that we only need to check certain keys of this two sections. While we still need to keep other sections, so we still need 621-622.

Reply:
Thanks for explanation.